[Wan Wei Kiat] iP

1waykiat opened the pull request:
# Duke 🙃 
> "You must be the change you wish to see in the world." - Gandhi

… and Duke allows you to accomplish exactly that. Leverage on the power of Duke to create the world you wish to see!

It is:
- Text-based
- Easy to learn
- ~~slow~~ ~~decent~~ ~~fast~~ **ULTRA SUPER FAST** to use

and guess what? 
😮 IT IS **FREE!!!** 😮 

## Features:

- [X] Manage your tasks
- [X] Mange your deadlines (coming ~~soon~~ **NOW**)
- [ ] Reminders (coming soon)
- [X] Micro-transactions (coming faster than you think 😈 )

##

Fellow Java programmers can use it to practice Java too! Here's the main method:
``` java
public static void main(String[] args) {
    new Duke("data/duke.txt").run();
}
```

Commit:
Add Gradle support

1waykiat committed:
Level-1

1waykiat committed:
Level-2

1waykiat committed:
Level-3

1waykiat committed:
Level-4

1waykiat committed:
Level-5

1waykiat committed:
Level-6

1waykiat committed:
Splitting all the classes to their individual files

1waykiat committed:
Cleaning up code

1waykiat committed:
Handling more errors (towards done)

1waykiat committed:
A-TextUiTesting

1waykiat committed:
Added documentation

1waykiat committed:
A-Enums

1waykiat committed:
changes made to input.txt and EXPECTED.TXT

1waykiat committed:
slight changes with Duke.java

1waykiat committed:
Level-7

1waykiat committed:
Level-8

1waykiat committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java

1waykiat committed:
A-MoreOOP

1waykiat committed:
A-Packages

1waykiat committed:
A-JUnit

1waykiat committed:
A-Jar

1waykiat committed:
A-JavaDoc

1waykiat committed:
Level-9

1waykiat committed:
A-CodingStandard

1waykiat committed:
Merge branch 'branch-A-JavaDoc'

1waykiat committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/general/Parser.java

1waykiat committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/general/Parser.java
#	src/main/java/duke/general/Ui.java

1waykiat committed:
Merge remote-tracking branch 'origin/add-gradle-support'

1waykiat committed:
A-Gradle

1waykiat committed:
A-CheckStyle

aryagiat commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,48 @@
+/**
+ * Chatbot that helps you form a task list
+ */
+
```
is this space necessary?

aryagiat commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,33 @@
+ */
+public class AddCommand extends Command {
+    private TaskType type;
+    private String[] input;
```
should this variable be plural form?

yeppog replied:
@aryagiat hmm are you referring to `input`? in this case, could we see it as a single input, and hence there is no need for plural forms to be used? 

technically a command would take in a single input at anytime. unless since you mean since it is `String[]` and hence we should take on plural forms

aryagiat commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,26 @@
+ * Command for the program to delete a task
+ */
+public class DeleteCommand extends Command {
+    private String[] input;
```
Same here and everywhere else, should it be in plural form?

yeppog commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,13 @@
+ * Class to deal with the actions and method calls to be done for different commands
+ */
+public abstract class Command {
+    public abstract void execute(Tasklist tasks, Storage storage, Ui ui) throws DukeException;
```
Since this is an abstract method, should we have some comments to describe what the method does to the parameters?

yeppog commented on src/main/java/duke/command/DoneCommand.java:
```diff
@@ -0,0 +1,27 @@
+
+    @Override
+    public void execute(Tasklist tasks, Storage storage, Ui ui) throws DukeException {
+        Task t = tasks.doneTask(input);
```
perhaps trivial, but could task `t` take on a more explicit name?

yeppog commented on src/main/java/duke/general/Tasklist.java:
```diff
@@ -0,0 +1,132 @@
+     * @return duke.task.Task that was added to the duke.general.Tasklist
+     */
+    public Task addTask(TaskType t, String[] inputSplit) throws DukeException {
+        Task temp;
```
quite a lot of duplicate code assigning `temp` to a new `Task` to add it to the list.
perhaps the command could handle the adding so that there is less duplicate code.

yeppog commented on src/main/java/duke/general/Tasklist.java:
```diff
@@ -0,0 +1,132 @@
+     * @return ArrayList of tasks that are related to the keyword
+     */
+    public ArrayList<Task> findTasks(String keyword) {
+        ArrayList<Task> found = new ArrayList<>();
```
should we use a name like `foundTasks` to be clear of the contents in the array, as well as to show that there could potentially be more than 1 item in `found`?

1waykiat committed:
Level-10

1waykiat committed:
Merge branch 'branch-Level-10'

meerian commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,174 @@
+        while (ui.getLoop()) {
+            try {
+                String fullCommand = ui.readInput();
+                Command c = parser.parse(fullCommand);
```
hey, I like how clean the run function of dukebot is 👍

meerian replied:
hey, I like how clean the run function of dukebot is 👍

meerian commented on text-ui-test/input.txt:
```diff
@@ -0,0 +1,18 @@
+done 3
+delete 1
+list
+bye
```
I like how rigorous your testing for text-ui-test is!

meerian reviewed (commented):
Good job overall with this project! Keep up the good work!👍

yeppog reviewed (commented):
good job!

1waykiat committed:
add FXML to manage JavaFX

1waykiat committed:
A-Assertions

1waykiat committed:
A-CodeQuality

1waykiat committed:
Merge pull request #1 from 1waykiat/branch-A-Assertions

A-Assertions

1waykiat committed:
Merge pull request #2 from 1waykiat/branch-A-CodeQuality

A-CodeQuality

1waykiat committed:
Merge branch 'master' of https://github.com/1waykiat/ip

1waykiat committed:
C-Undo

1waykiat committed:
Change implementation of Undo

The undo implementation for add / delete was previously different from the done, where save states are used in the former
and manually the completion of tasks in the latter.

Added a method to create deepcopy of arraylist of task, and set a save state for all 3 commands instead.

Standardises the method of saving states throughout the app.

1waykiat committed:
Finalise code styling

Tidying up code and added documentation for certain methods that were left blank previously.

1waykiat committed:
A-UserGuide

1waykiat committed:
Set theme jekyll-theme-slate

1waykiat committed:
Set theme jekyll-theme-cayman

1waykiat committed:
Minor change to README.md

1waykiat committed:
Merge branch 'master' of https://github.com/1waykiat/ip

over-fitted commented on docs/README.md:
````diff
@@ -1,29 +1,153 @@
+Now you have 6 tasks in your list.
+```
+
+###Adding Deadline Type
````
Space should be added after ### lest your markdown not be rendered correctly

1waykiat committed:
Finalising

1waykiat committed:
Delete ui.png

Deleting Ui pic to replace with another

1waykiat committed:
Add Ui.png to docs

for A-UserGuide
